src/app/components/UserEventDetails.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import type { IEvent } from "../../database/eventSchema";
+import style from "@styles/UserEventDetails.module.css";
+
+type IParams = {
+  id: string;
+};
+
+export default function UserEventDetails({ id }: IParams) {
+  const [eventData, setEventData] = useState<IEvent | null>(null);
+
+  async function fetchEventData() {
+    try {
+      const response = await fetch(`http://localhost:3000/api/event/${id}`);
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch event. Status: ${response.status}`);
+      }
+
+      const data = await response.json();
+      setEventData(data);
+    } catch (err: unknown) {
+      console.error("Error:", err);
+      setEventData(null);
```

Comment:
If error, set data to null so the Event not found shows up instead of just keeping whatever was displayed previously.